### PR TITLE
settlers can rename their pets

### PR DESCRIPTION
**Pull request by Ben10Omintrix**
## About The Pull Request
settlers can now rename their tamed pets

## Why It's Good For The Game
adds another fun and harmless interaction between settlers and their pets

## Changelog
:cl:
add: settlers can rename their pets
/:cl:

### CODE REVIEW

**Comment by Justice12354** [comment]
Waiting for someone to call their pet "your mom" and say "I'm riding your mom"

File: code/datums/components/tameable.dm [hunk 1]
```diff
@@ -70,18 +70,28 @@
...
+	if(HAS_TRAIT(tamer, TRAIT_SETTLER))
+		INVOKE_ASYNC(src, PROC_REF(rename_pet), source, tamer)
 	if(unique)
 		qdel(src)
 	else
 		current_tame_chance = tame_chance
+
+/datum/component/tameable/proc/rename_pet(mob/living/animal, mob/living/tamer)
+	var/chosen_name = sanitize_name(tgui_input_text(tamer, "Choose your pet's name!", "Name pet", animal.name, MAX_NAME_LEN), allow_numbers = TRUE)
+	if(isnull(animal))
```
Comment (Ghommie) [inline code/datums/components/tameable.dm:91]:
```suggestion
	if(QDELETED(animal) || chosen_name == animal.name)
```

File: code/datums/components/tameable.dm [hunk 2]
```diff
@@ -70,18 +70,28 @@
...
+		return
+	if(!chosen_name)
+		to_chat(tamer, span_warning("Please enter a valid name."))
+		rename_pet(animal, tamer)
+		return
+	animal.name = chosen_name
```
Comment (Ghommie) [inline code/datums/components/tameable.dm:97]:
```suggestion
	animal.fully_replace_character_name(name, chosen_name)
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/datums/components/tameable.dm (hunk 1 above)
Comment (Ben10Omintrix) [reply code/datums/components/tameable.dm:91]: Are u sure about this? I want to give players the option to skip renaming their pets if they don't want to change it

Comment (Ghommie) [reply code/datums/components/tameable.dm:91]: Yes, this simply perform an early return if the animal is gone or if the animal name was not changed.

File: code/datums/components/tameable.dm [hunk 3]
```diff
@@ -70,18 +70,28 @@
...
+
+/datum/component/tameable/proc/rename_pet(mob/living/animal, mob/living/tamer)
+	var/chosen_name = sanitize_name(tgui_input_text(tamer, "Choose your pet's name!", "Name pet", animal.name, MAX_NAME_LEN), allow_numbers = TRUE)
+	if(isnull(animal))
+		return
+	if(!chosen_name)
+		to_chat(tamer, span_warning("Please enter a valid name."))
+		rename_pet(animal, tamer)
+		return
+	animal.fully_replace_character_name(name, chosen_name)
```
Comment (Ghommie) [inline code/datums/components/tameable.dm:97]:
```suggestion
	animal.fully_replace_character_name(animal.name, chosen_name)
```
My bad.

File: code/datums/components/tameable.dm (hunk 1 above)
Comment (Ben10Omintrix) [reply code/datums/components/tameable.dm:91]: oh yeah my bad i was looking at the code on mobile

**Comment by MTandi** [comment]
I thought of implementing this, but I've been thinking to use a collar item for this. This would make sense why everyone else see the given name.

Please consider this idea.

**Pull request merged**